src/plugins/analytics.js

```diff
@@ -0,0 +1,54 @@
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment'
+
+export default (function () {
+  if (!ExecutionEnvironment.canUseDOM) {
+    return null
+  }
+
+  // Emit events for all github link clicks
+  function anchorListener(event) {
+    let element = event.target
+    if (
+      element.tagName == 'A' &&
+      element.href.startsWith('https://github.com')
+    ) {
+      analytics.track('Github Navigation', {
+        type: document.title.split(' | ')[0],
+        url: element.href,
+        text: element.textContent,
+      })
+    }
+  }
+
+  function recordDocumentationPage() {
+    const categoryElement = document.querySelector(
+      '.menu__link--sublist.menu__link--active',
+    )
+
+    const titleElement = document.querySelector('.menu__link--active.active')
+
+    if (categoryElement && titleElement) {
+      analytics.track('Documentation Viewed', {
+        category: categoryElement.textContent,
+        title: titleElement.textContent,
+      })
+    } else if (titleElement) {
+      analytics.track('Documentation Viewed', {
+        title: titleElement.textContent,
+      })
+    }
+  }
+
+  // On any click call the anchor listener event
+  document.addEventListener('click', anchorListener)
+
+  // Wait until next tick to record documentation viewed
+  setTimeout(recordDocumentationPage, 100)
```

Comment:
Do we want to record this for the new docs landing page which is a marketing page?

Reply:
Not sure on that, that would depend on what the data team wants. I'll reach out on that.